change/@ni-nimble-angular-e488db99-92c5-4954-a624-673ac1c31f32.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Why are these not issues for Angular? Is there something unique to us that is causing it to be a problem?

Why is it intermittent? How confident are we that it won't be a whack-a-mole?

Reply:
Good questions. When we first forked that test file, we got an error about `AbsoluteSimpleLinkCmp` and `RelativeLinkCmp` potentially generating colliding IDs. I was surprised that Angular didn't have to work around it, too. It would take time to investigate, and I'm not sure it's worth it right now. Apparently, the issue is limited to those two components. This is an easy workaround to try, and if we never run into the error again, great. If we do, we can spend a little more time on it.

Reply:
I mean with that reasoning (we don't know why it's intermittent and we don't want to investigate unless it is frequently intermittent) then why is there any confidence in making the current change? Doesn't the same logic apply (ignoring the intermittent test policy for this specific case)?

Do the tests fail consistently without the "irrelevant :not(head)" selector? Maybe we should remove all of them to better characterize the intermittency? Maybe we should add them to all of them?

Might be worth a post stand-up discussion. Would like @jattasNI's take on it as well.

Reply:
Apparently Angular is setting the global property `ngDevMode` to `false` for their test runs. I couldn't find exactly where in their build/test infrastructure this is happening, but I'm very confident. The ID collision error is [only reported when `ngDevMode` is undefined or truthy](https://github.com/angular/angular/blob/ace0a683f60eba3d30d11f241c59e37386b2b805/packages/core/src/render3/definition.ts#L723). By adding `window.ngDevMode = false;` or `enableProdMode()` to the test file, I am able to avoid the ID collision error, but it interferes with other tests being run. They fail with errors like `Cannot redefine property: ɵfac` and `Cannot configure the test module when the test module has already been instantiated. Make sure you are not using ``inject`` before ``TestBed.configureTestingModule``.`. If I add an `afterAll()` that sets `ngDevMode` back to undefined, it reduces the number of failures down to a handful, but doesn't fully solve the problem.

It's also not fully clear to me if there are any potential downsides to running in production mode. Given that we now have an explanation for why the issue is limited to just a couple of the components in the test file (see my response in the other thread), I think the change in this PR is probably our best option.

Reply:
Why tests are intermittent is still a good question but the overview given at stand-up and here gives more clarity into why we see the issues and Angular tests don't. We can follow the suggestion for the updated tweak in the PR 👍